src/Compilers/CSharp/Portable/Symbols/NamespaceOrTypeSymbol.cs

```diff
@@ -326,6 +330,34 @@ internal virtual NamedTypeSymbol LookupMetadataType(ref MetadataTypeName emitted
             }
 
 Done:
+            if (isTopLevel
```

Comment:
I agree that handling lookup for error cases quickly gets complicated :-/ Not blocking this PR, but it would be good to check what Aleksey thinks on supporting "correct" behavior of public APIs in error cases.

Reply:
@AlekseyTs The scenario is `GetTypeByMetadataName` on a nested file-type (`class C { file class D { } }`), ie. an error scenario.

Reply:
> The scenario is ```GetTypeByMetadataName``` on a nested file-type (``class C { file class D { } }```), ie. an error scenario.

I think in this case the file modifier on a nested type should be ignored and class D should be treated as a regular nested type. In other words, I think that the best implementation strategy here will be to not have nested file types in the symbol model, under no condition.

Reply:
Perfect. That's the approach Rikki took and it makes things simpler. Thanks

@RikkiGibson If we take that approach, let's verify the symbol for class `D` has `!IsFile` then.